r/R/dplyr-functions.R

```diff
@@ -0,0 +1,352 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
I'm not totally convinced by this myself, but another option for the name of this file could be `nse-functions` which matches the object we put them in and is slightly broader than dplyr only. Though that might hurt discoverability since I think ~everyone looking at this code will associate these with dplyr and be looking for that in the file name

Reply:
Yeah I want to keep all of the dplyr related things namespaced (such as we can inside a flat `R/` directory)